x-pack/plugins/security_solution/public/management/pages/endpoint_hosts/view/hooks/hooks.ts

```diff
@@ -55,3 +59,74 @@ export const useAgentDetailsIngestUrl = (
     };
   }, [services.application, agentId]);
 };
+
+export const useLogEntryUIElements = (logEntry: Immutable<ActivityLogEntry>) => {
+  const iconType = useRef<IconType>('dot');
+  const commentType = useRef<EuiCommentProps['type']>('update');
+  const commentText = useRef<string | undefined>();
+  const avatarSize = useRef<EuiAvatarProps['size']>('s');
+  const isIsolateAction = useRef<boolean>(false);
+  const isResponseEvent = useRef<boolean>(false);
+  const isSuccessful = useRef<boolean>(false);
+  const displayComment = useRef<boolean>(false);
+  const displayResponseEvent = useRef<boolean>(true);
+  const username = useRef<EuiCommentProps['username']>('');
+
+  return useMemo(() => {
+    if (logEntry.type === 'action') {
+      avatarSize.current = 'm';
+      commentType.current = 'regular';
+      commentText.current = logEntry.item.data.data.comment ?? '';
+      displayResponseEvent.current = false;
+      iconType.current = 'lockOpen';
+      username.current = logEntry.item.data.user_id;
+      if (logEntry.item.data.data) {
+        const data = logEntry.item.data.data;
+        if (data.command === 'isolate') {
+          iconType.current = 'lock';
+          isIsolateAction.current = true;
+        }
+        if (data.comment) {
+          displayComment.current = true;
+        }
+      }
+    } else if (logEntry.type === 'response') {
+      isResponseEvent.current = true;
+      if (logEntry.item.data.action_data.command === 'isolate') {
+        isIsolateAction.current = true;
+      }
+      if (!!logEntry.item.data.completed_at && !logEntry.item.data.error) {
+        isSuccessful.current = true;
+      }
+    }
+
+    const actionEventTitle = `${
+      isIsolateAction.current
+        ? i18.ACTIVITY_LOG.LogEntry.action.isolated
+        : i18.ACTIVITY_LOG.LogEntry.action.unisolated
+    } ${i18.ACTIVITY_LOG.LogEntry.host}`;
+    const responseEventTitle = `${i18.ACTIVITY_LOG.LogEntry.host}  ${
```

Comment:
You should not do this - create a string that concatenates multiple i18n values instead, you should create a new i18n value that accepts substitutable tokens. like:

```typescript
i18n.translate('id...', {
  defaultMessage: '{host} {actionStatus}',
  values: {
    host: 'some value',
    actionStatus: 'some value'
  }
});
```

But even this I don't think is good - to use i18n values as replaceable token values.

Reply:
Ah, I see. Why shouldn't we concatenate i18n values? Perhaps I should just wrap the text in i18n methods right where they are instead of substituting values? 🤔

Reply:
Concatenating multiple i18n value will make it impossible for translators to ensure the output makes sense in the desired language, especially when the order of words may have to be flipped or additional context added to the message.

Yes, you could have a function defined in the translations file that accepts the arguments you want to pass along to the `i18n.translate()`.

Reply:
So here c4e933a9c5954ce249942ca66bab380c1dfa79e2 I've added the sentences for translation instead, which I believe is better than substituting translated words like you pointed out.